### PR TITLE
[WebNN EP] Support Where Op

### PR DESCRIPTION
**Pull request by zesongw**
### Description
Add Where Op for WebNN EP as ternary conditional operator.

### CODE REVIEW

**Comment by zesongw** [comment]
@guschmue, @fdwr, PTAL, thanks!

**Comment by fdwr** [comment]
Cool, looking now... (I started a PR for `elementwiseIf` but never published it :b).

File: onnxruntime/core/providers/webnn/builders/impl/ternary_op_builder.cc [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  Status AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node,
+                               const logging::Logger& logger) const override ORT_MUST_USE_RESULT;
+};
+
+// Add operator related.
+
+Status TernaryOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node,
+                                              const logging::Logger& /* logger */) const {
+  const auto& op_type(node.OpType());
+
```
Comment (fdwr) [inline onnxruntime/core/providers/webnn/builders/impl/ternary_op_builder.cc:28]:
```suggestion
  ORT_RETURN_IF(node.InputDefs().size() < 3, "Operator requires at least three inputs");
  
```

I'd just add a safety check here.

**Review by fdwr — COMMENTED** [review]
One request. Otherwise looks good.

**Review by fdwr — DISMISSED** [review]
👍 Thanks Zesong.

File: onnxruntime/core/providers/webnn/builders/impl/ternary_op_builder.cc [hunk 2]
```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
[the remark this answers is not in the document]
Comment (guschmue) [reply onnxruntime/core/providers/webnn/builders/impl/ternary_op_builder.cc:1]: ternary_op_builder.cc:25

the details are here: 
https://github.com/microsoft/onnxruntime/actions/runs/5293356510/jobs/9581385729?pr=16380

[the remark this answers is not in the document]
Comment (fdwr) [reply onnxruntime/core/providers/webnn/builders/impl/ternary_op_builder.cc:1]: Evidently it's complaining that the line is longer than 120 characters (well at least in ORT, we're not stuck in the dinosaur days of VGA monitors with 80 columns like TensorFlow and Chrome, which really hinder readability when browsing the logic).

image

**Review by fdwr — DISMISSED** [review]

**Review by fdwr — APPROVED** [review]

**Comment by fdwr** [comment]
Restarting random failed job 🤨.

```
Windows GPU CI Pipeline
build_x64_RelWithDebInfo

CMake Error at CMakeLists.txt:33 (project):
  Generator

    Visual Studio 17 2022

  could not find any instance of Visual Studio.
```

**Review by guschmue — APPROVED** [review]

**Pull request merged**